### PR TITLE
Update code standards

### PR DESCRIPTION
**Pull request by JacobCallahan**
This pr introduces some new changes to this project's code standards.
Highlighted changes revolve around string formatting, fixture naming,
string concatenation, and black.

### CODE REVIEW

**Review by mshriver — APPROVED** [review]

**Review by swadeley — APPROVED** [review]
ACK

File: docs/code_standards.rst
Comment (jaryn) [inline docs/code_standards.rst:19]:
```suggestion
* Robottelo uses the code formatting tool: black.
```

Comment (swadeley) [reply docs/code_standards.rst:19]: As its not a list, I disagree to the use of a colon. Some style guides suggest italics for the first use of a key term, or use quotation marks in a man mage.

Looking at https://github.com/psf/black I see the author uses a capital letter to write Black.

Comment (jaryn) [reply docs/code_standards.rst:19]: OK. Whatever makes it more obvious it is a name of the tool.

Comment (mshriver) [reply docs/code_standards.rst:19]: A link here would probably be beneficial to anyone new to the tool.

**Comment by mshriver** [comment]
Travis black failure will be resolved by #8131

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:19]: @swadeley thoughts?

**Comment by JacobCallahan** [comment]
@SatelliteQE/robottelo-tier-1-reviewers @SatelliteQE/airgun-tier-2-reviewers please review and let me know your thoughts. I'd like to have most reviewers represented in this PR.

File: docs/code_standards.rst
Comment (swadeley) [reply docs/code_standards.rst:19]: Looking at the rendered guide[1], making the word "Black" a link to https://github.com/psf/black solves this issue as the word will be marked up in a blue font. No need for italics then.

[1] https://robottelo.readthedocs.io/en/latest/code_standards.html

**Review by omkarkhatavkar — APPROVED** [review]
ACK

File: docs/code_standards.rst
Comment (latran) [inline docs/code_standards.rst:96]: In some editors or IDEs, you can format the indent of your tab to be a certain number of spaces.  Correct me if I'm wrong, but it seems pycharm uses 4 spaces as default for it's tabs.

**Review by latran — APPROVED** [review]
ACK

**Review by ColeHiggins2 — APPROVED** [review]
ACK

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:96]: that is correct. for most people, setting this as their editor default is the best behavior. black might also correct for this.

File: docs/code_standards.rst
Comment (lhellebr) [inline docs/code_standards.rst:11]:
```suggestion
contributors are held to. Violations to our strictly held standards will result in a rejected pull request until all violations have been resolved. While not adhering to our recommended standards isn't a show stopper, keeping to them will help our code base to stay great!
```

**Review by lhellebr — APPROVED** [review]

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:11]: ahh nice catch!

File: docs/code_standards.rst
Comment (pondrejk) [inline docs/code_standards.rst:66]: why not 
```
'this {0} is a {0} that has the {1} {0} {2}'.format('sentence', 'word', 'multiple times')
```

Comment (JacobCallahan) [reply docs/code_standards.rst:66]: Would people prefer using indices of kwargs here?

Comment (jyejare) [reply docs/code_standards.rst:66]: The indices works well (look up) when we have less number of arguments to format but when the arguments count is high the kwargs works better.

If the number of code lines doesnt matter. then I would go with kwargs.

**Review by jameerpathan111 — APPROVED** [review]
LGTM

File: docs/code_standards.rst
Comment (jyejare) [inline docs/code_standards.rst:92]: Should we altogether remove it as black and pre-commit take cares of it automatically about styling.

File: docs/code_standards.rst
Comment (jyejare) [inline docs/code_standards.rst:118]: I think `+` concatenation way is always consider bad way of doing it. Can we altogether disagree with `+` concatenation.

fstings solved it much better.

**Review by jyejare — COMMENTED** [review]
ACK with non-blocking / improvement comments.

Sorry for the delayed response on the PR. Nice additions to the code_standards overall.

File: docs/code_standards.rst
Comment (mshriver) [reply docs/code_standards.rst:118]: I agree with you here @jyejare but was willing to accept it, as @JacobCallahan had some justification for performance.

Comment (jyejare) [reply docs/code_standards.rst:118]: @mshriver Not exactly sure how much time it would really save (probably in milliseconds), but f-string has an upper-hand to plus concatenation when it comes to readability/ease.

File: docs/code_standards.rst
Comment (pondrejk) [reply docs/code_standards.rst:66]: I guess we won't see it very often, maybe we can leave it to contributor's discretion.

**Review by pondrejk — APPROVED** [review]

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:22]: I'll clarify the language to indicate that the CI checks that the code is blak-compliant.

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:86]: "as" is correct here. Also, this section is about naming convention rather than any kind of pytest specification.

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:90]: This section is about how to name fixtures so their scope is apparent, not to give instruction on defining fixture scope.

With that, I'm not a fan of the default scope "function" being explicitly included in the fixture name by prefixing it with "func_"

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:92]: I can remove this

File: docs/code_standards.rst
Comment (JacobCallahan) [reply docs/code_standards.rst:118]: @jyejare I'll leave the guidance here as-is, since trivial string concatenation with `+` is both easier to write/read and more performant that f-strings.

**Comment by mshriver** [comment]
I believe all comments have been addressed, and we have sufficient approvals from the reviewer team members. Any further changes to the coding standards can be addressed as individual PRs.  Thanks for advancing our standards @JacobCallahan !

**Pull request merged**
